TESTDATA_GUIDELINES.md

```diff
@@ -0,0 +1,277 @@
+the UI, or perhaps the coordinator gave th wrong instruction), the coordinator also writes this down.
+
+At first glance it might seem overkill to have two people 'just play around with a phone', but a mistake is made **very
+easily**. Having two people in this process greatly reduces the number of mistakes you will make (yes, mistakes will be
+made).
+
+By having the coordinator annotate everything, you end up with a complete and correct script of all actions that were
+taken, including the time stamps. This is valuable metadata that you can store along your reference data after
+extraction, so when you analyze the reference data, you know exactly how it was created and what you could find in that
+data.
+
+## 3. Extracting and storing the test data
+
+### Extraction
+
+After you have executed the script, your reference data is now ready, you just need to extract it. How to do this
+depends on what data you need:
+
+1. If you do network analysis, I hope you didn't forget to record the network traffic!
+2. If you want a full device image, use your favorite tool to make this image
+3. If you only want specific app data, you'll need to delve into the device to find its location
+
+The first two cases are trivial, in the sense that if you're doing this sort of analysis you probably don't need any
+help to collect the data you want.
+
+In the last case, you might not know where to get the data. And you;re not alone: I don't know either. Where app store
+their data is dependent on the OS and the app itself.
+
+Luckily on modern mobile devices, apps cannot just store data anywhere they want. On android, relevant app data is
+usually stored in `/data/data/<package name of your app>`. Within this folder, the app can chose how to store its data.
+What we usually do is copy this entire folder and then explore this data.
+
+### Storing the test data
+
+After you extracted the data you want, you can now do your thing with it, but after that, you probably don't want to
+delete it.
+
+At the NFI, we store all reference data, along with metadata describing the reference data. This metadata file includes
+the following:
+
+* file type + version if applicable
+* Creation date + location
+* Device + OS on which it was created
+* Content description (eg `chat database of foo@bar.com, containing 213 messages`)
+* Reproduction steps: the exact and complete steps you took to create this data. This contains either a reference to
+  your Puma script, or the annotated script
+
+By storing these together, you can start building a full archive of high-quality reference data of which anyone can
+easily find out where data came from and how it was created. If anyone in the (far) future ever wants to reproduce your
+process, or expand on it, they can now easily do so.
+
+## 4. Conclusion
+
+The creation of test data is not a trivial thing, and doing it right even less so. In this document I tried to summarize
```

Comment:
I would maybe avoid using "I" in this text. It feels a little too informal. I think it's possible to get the same point across without using it?